apps/ui/src/views/My/Notifications.vue

```diff
@@ -0,0 +1,69 @@
+<script setup lang="ts">
+import { _rt } from '@/helpers/utils';
+
+const notificationsStore = useNotificationsStore();
+const { setTitle } = useTitle();
+
+watchEffect(async () => {
+  setTitle(
+    `Notifications${notificationsStore.unreadNotificationsCount ? ` (${notificationsStore.unreadNotificationsCount} unread)` : ''}`
+  );
+});
+
+watch(
+  () => notificationsStore.unreadNotificationsCount,
+  () => {
+    notificationsStore.refreshLastUnreadTs();
+  },
+  { immediate: true }
+);
+
+onUnmounted(() => notificationsStore.markAllAsRead());
```

Comment:
Seems to be working well now, I wonder, would it make sense to remove `markAllAsRead` entirely and instead of doing this:
```js
unread: timestamp > shownLastUnreadTs.value
```

for each proposal when those are fetched we make it like this:

```js
const unreadNotificationsCount = computed(
  () => notifications.value.filter(n => n.timestamp > shownLastUnreadTs.value).length
);
```

This way we don't need `markAllAsRead` call at all, as it's unreliable anyway.

Reply:
We would still need the `markAllAsRead` to mark all notifications as read, when user is navigating away from notification page